src/utils/flashMintNotional/notional.ts

```diff
@@ -0,0 +1,167 @@
+import { useCallback, useState } from 'react'
+
+import { Signer } from 'ethers'
+
+import { BigNumber } from '@ethersproject/bignumber'
+
+import { useBalances } from 'hooks/useBalance'
+import { FlashMintNotionalQuote } from 'hooks/useBestQuote'
+import { useNetwork } from 'hooks/useNetwork'
+import { useWallet } from 'hooks/useWallet'
+import { fromWei } from 'utils'
+import { logTx } from 'utils/api/analytics'
+import {
+  CaptureExchangeIssuanceFunctionKey,
+  CaptureExchangeIssuanceKey,
+  captureTransaction,
+} from 'utils/api/sentry'
+import { getAddressForToken } from 'utils/tokens'
+
+import { getFlashMintNotionalContract } from './fmNotionalContract'
+import {
+  FlashMintNotionalGasEstimateFailedError,
+  getFlashMintNotionalGasEstimate,
+} from './fmNotionalGasEstimate'
+
+export const useTradeFlashMintNotional = () => {
```

Comment:
This seems like it'd be more appropriately labeled a react hook than a utility. If it is, shouldn't we consider adding this into the hooks folder and treating this module as a react hook.